aesd-char-driver/aesd-circular-buffer.c

```diff
@@ -96,3 +110,25 @@ void aesd_circular_buffer_init(struct aesd_circular_buffer *buffer)
     buffer->full =false;
 
 }
+
+void aesd_circular_buffer_free(struct aesd_circular_buffer *buffer)
+{
+
+    uint8_t index;
+    uint8_t temp = 0;
+
+   for(index = buffer->out_offs; temp < AESDCHAR_MAX_WRITE_OPERATIONS_SUPPORTED ; index = (index +1) % AESDCHAR_MAX_WRITE_OPERATIONS_SUPPORTED)
+    {
+        ++temp;
+        if( buffer->entry[index].buffptr != NULL)
+        {
+
+            #ifdef __KERNEL__
+                kfree(buffer->entry[index].buffptr);
```

Comment:
@Sharath-Jonnala I'd suggest moving this into the application code, since it will know how the buffer is allocated.  The code below won't work with unit test since buffers aren't dynamically allocated.